### PR TITLE
add promptflow code deploy samples

### PR DESCRIPTION
**Pull request by likebupt**
# Description

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: cli/generative-ai/promptflow/basic-chat/data.jsonl
Comment (cloga) [inline cli/generative-ai/promptflow/basic-chat/data.jsonl:1]: Do we need put any data here?

File: cli/generative-ai/promptflow/basic-chat/flow.dag.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    question: ${inputs.question}
+  name: chat
+  type: llm
+  source:
+    type: code
+    path: chat.jinja2
+  api: chat
+  provider: AzureOpenAI
+  connection: azure_open_ai_connection
+node_variants: {}
```
Comment (cloga) [inline cli/generative-ai/promptflow/basic-chat/flow.dag.yaml:25]: We need add environment part and requirements.txt

File: cli/generative-ai/promptflow/deploy-flow/deployment-with-docker-context-environment.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+instance_count: 1
+environment_variables:
+  # When there are multiple fields in the response, using this env variable will filter the fields to expose in the response.
+  # For example, if there are 2 flow outputs: "answer", "context", and I only want to have "answer" in the endpoint response, I can set this env variable to '["answer"]'
+  # PROMPTFLOW_RESPONSE_INCLUDED_FIELDS: '["category", "evidence"]'
+
+  # if you want to deploy to serving mode, you need to set this env variable to "serving"
+  PROMPTFLOW_RUN_MODE: serving
+  # currently it is for pulling connections from workspace
+  PRT_CONFIG_OVERRIDE: deployment.subscription_id=<subscription_id>,deployment.resource_group=<resource_group_name>,deployment.workspace_name=<workspace_name>,deployment.endpoint_name=basic-chat-endpoint,deployment.deployment_name=basic-chat-custom-docker
```
Comment (cloga) [inline cli/generative-ai/promptflow/deploy-flow/deployment-with-docker-context-environment.yaml:40]: Need mark real value.

File: cli/generative-ai/promptflow/deploy-flow/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+instance_count: 1
+environment_variables:
+  # When there are multiple fields in the response, using this env variable will filter the fields to expose in the response.
+  # For example, if there are 2 flow outputs: "answer", "context", and I only want to have "answer" in the endpoint response, I can set this env variable to '["answer"]'
+  # PROMPTFLOW_RESPONSE_INCLUDED_FIELDS: '["category", "evidence"]'
+
+  # if you want to deploy to serving mode, you need to set this env variable to "serving"
+  PROMPTFLOW_RUN_MODE: serving
+  # currently it is for pulling connections from workspace
+  PRT_CONFIG_OVERRIDE: deployment.subscription_id=ee85ed72-2b26-48f6-a0e8-cb5bcf98fbd9,deployment.resource_group=keli19-aml,deployment.workspace_name=keli19-eastus,deployment.endpoint_name=basic-chat-endpoint-1025,deployment.deployment_name=basic-chat-monitor
```
Comment (cloga) [inline cli/generative-ai/promptflow/deploy-flow/deployment.yaml:29]: Need mast real values.

**Review by cloga — APPROVED** [review]

File: cli/generative-ai/promptflow/basic-chat/data.jsonl
Comment (likebupt) [reply cli/generative-ai/promptflow/basic-chat/data.jsonl:1]: no, this data file is not used in deployment, I'll delete it

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**